src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala

```diff
@@ -9,90 +9,216 @@ import org.apache.spark.ml.param.DataFrameEquality
 import org.apache.spark.sql.{DataFrame, Row}
 import org.apache.spark.sql.functions.col
 import com.microsoft.ml.spark.stages.FixedMiniBatchTransformer
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
+import org.scalatest.exceptions.TestFailedException
 
 
+class DetectedLanguageSuiteV4 extends TestBase with DataFrameEquality with TextKey {
 
-class DetectedLanguageSuitev4 extends TestBase with DataFrameEquality with TextKey {
   import spark.implicits._
+
   lazy val df: DataFrame = Seq(
     "Hello World",
     "Bonjour tout le monde",
     "La carretera estaba atascada. Había mucho tráfico el día de ayer.",
     "世界您好",
     ":) :( :D",
-  ).toDF("text2")
+  ).toDF("In - Language")
 
   val options: Option[TextAnalyticsRequestOptions] = Some(new TextAnalyticsRequestOptions()
     .setIncludeStatistics(true))
 
   lazy val detector: TextAnalyticsLanguageDetection = new TextAnalyticsLanguageDetection(options)
     .setSubscriptionKey(textKey)
     .setEndpoint("https://eastus.api.cognitive.microsoft.com/")
-    .setInputCol("text2")
+    .setInputCol("In - Language")
+    .setOutputCol("Out - Language")
 
   test("Language Detection - Basic Usage") {
-    val replies = detector.transform(df)
-      .select("name", "iso6391Name")
+    val outputCol = detector.transform(df)
+      .select("Out - Language")
       .collect()
+    assert(outputCol(0).schema(0).name == "Out - Language")
+
+    outputCol.foreach(row => {
+      var outResponse: GenericRowWithSchema = row.getAs[GenericRowWithSchema]("Out - Language")
+
+      val result = outResponse.getAs[GenericRowWithSchema]("result");
+      val error = outResponse.getAs[GenericRowWithSchema]("error");
+      val statistics = outResponse.getAs[GenericRowWithSchema]("statistics");
+      val modelVersion = outResponse.getAs[String]("modelVersion");
+      val language = result.getAs[String]("name")
+      val langCode = result.getAs[String]("iso6391Name")
+      val confidence = result.getAs[Double]("confidenceScore")
+
+      assert(modelVersion == "2021-01-05")
+
+      if (language == "English") {

```

Comment:
You should rarely have any `if` statements in tests. It essentially means the test is not deterministic which is a problem.
Every test should have clear input, and clear set of output expectations. For example, first line should be English, second French, etc.